### PR TITLE
Fix performance issue in address page

### PR DESCRIPTION
**Pull request by fvictorio**
The address page was taking a long time to load, and the server was failing with a 504.

The slow query was composed of several `where`s and limiting the result to the page size (51). For some reason, running all of the filters together was very slow. This solutions runs a separate query for each filter, limiting the result to the same number. Then, the results are merged, sorted and limited again. This is done in the backend because there are few rows involved, but it could be done using `UNION` and an outer query too.

Notice that the queries are done sequentially. This could be further improved if each query is done in parallel.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain.ex [hunk 1]
```diff
@@ -187,18 +187,45 @@ defmodule Explorer.Chain do
 
     token_transfers_dynamic = TokenTransfer.dynamic_any_address_fields_match(direction, address_bytes)
 
-    transaction_dynamic =
-      Transaction.dynamic_where_address_hash_matches(address_hash, direction, token_transfers_dynamic)
```
Comment (amandasposito) [inline apps/explorer/lib/explorer/chain.ex]: You can go ahead and ✂️ this function from the `Transaction` module, this function is only used here.

File: apps/explorer/lib/explorer/chain.ex [hunk 2]
```diff
@@ -187,18 +187,45 @@ defmodule Explorer.Chain do
...
+
+    queries =
+      [token_transfers_query] ++
+        case direction do
+          :from -> [from_address_query]
+          :to -> [to_address_query, created_contract_query]
+          _ -> [from_address_query, to_address_query, created_contract_query]
+        end
+
+    result = queries |> Enum.flat_map(&Repo.all/1)
```
Comment (amandasposito) [inline apps/explorer/lib/explorer/chain.ex]: No need to use pipe just once 🙂

https://github.com/rrrene/credo/issues/221

**Comment by amandasposito** [comment]
I tested it using the mainnet db test and everything seems to be working fine, just solve the comments above and I think we are good to go 👍

File: apps/explorer/lib/explorer/chain.ex (hunk 1 above)
Comment (fvictorio) [reply apps/explorer/lib/explorer/chain.ex]: Done, thanks :+1:

File: apps/explorer/lib/explorer/chain.ex (hunk 2 above)
Comment (fvictorio) [reply apps/explorer/lib/explorer/chain.ex]: Fixed. thanks. Btw, I didn't get a warning when I ran credo, and it didn't trigger a warning in the CI. Is this something that needs to be configured?

Comment (amandasposito) [reply apps/explorer/lib/explorer/chain.ex]: I was also wondering why this didn't trigger a warning, I need to check the credo config.

**Pull request merged**